client/lib/upgrades/actions/domain-management.js

```diff
@@ -296,6 +296,12 @@ function applyDnsTemplate( domainName, template, variables, onComplete ) {
 	} );
 }
 
+function getDnsTemplateConflicts( domainName, template, variables, onComplete ) {
```

Comment:
Yep. I actually put both `wpcom.getDnsTemplateConflicts` and `wpcom.applyDnsTemplate` directly in the component with promises.  ...much better!